Objective-C/CBLDatabase.mm

```diff
@@ -178,12 +178,16 @@ - (NSString*) description {
 
 
 - (NSString*) path {
-    return _c4db != nullptr ? sliceResult2FilesystemPath(c4db_getPath(_c4db)) : nil;
+    @synchronized (self) {
```

Comment:
I understand that it behaves the same. What is the benefit of using a mutex object instead of self?